src/game/client/draw/w3dlaserdraw.cpp

```diff
@@ -13,3 +13,314 @@
+                    c1.Set(&end);
+                    c1.Sub(&start);
+                    float f1 = c1.Length();
+
+                    Coord3D c2;
+                    c2.Set(&start);
+                    c2.Add(&end);
+                    c2.Scale(0.5f);
+
+                    float f2 = f1 * 0.5f;
+                    float f3 = (float)i / (float)data->m_segments;
+                    float f4 = ((float)i + 1.0f) / (float)data->m_segments;
+
+                    if (i > 0) {
+                        f3 = f3 - data->m_segmentOverlapRatio;
+                    }
+
+                    if (i < data->m_segments - 1) {
+                        f4 = f4 + data->m_segmentOverlapRatio;
+                    }
+
+                    Coord3D c3;
+                    c3.Set(&c1);
+                    c3.Scale(f3);
+
+                    Coord3D c4;
+                    c4.Set(&start);
+                    c4.Add(&c3);
+
+                    c3.Set(&c1);
+                    c3.Scale(f4);
+
+                    Coord3D c5;
+                    c5.Set(&start);
+                    c5.Add(&c3);
+
+                    c3.Set(&c2);
+                    c3.Sub(&c4);
+
+                    float f5 = c3.Length();
+                    float f6 = f5 / f2 * GAMEMATH_PI * 0.5f;
+                    float f7 = GameMath::Cos(f6);
+                    f7 = f7 * data->m_arcHeight;
+                    c4.z = c4.z + f7;
+
+                    c3.Set(&c2);
+                    c3.Sub(&c5);
+                    float f8 = c3.Length();
+                    float f9 = f8 / f2 * GAMEMATH_PI * 0.5f;
+                    f7 = GameMath::Cos(f9);
+                    f7 = f7 * data->m_arcHeight;
+                    c5.z = c5.z + f7;
+
+                    if (g_theTerrainLogic->Get_Ground_Height(c4.x, c4.y, nullptr) + 2.0f >= c4.z) {
+                        points[0].Set(c4.x, c4.y, g_theTerrainLogic->Get_Ground_Height(c4.x, c4.y, nullptr) + 2.0f);
+                    } else {
+                        points[0].Set(c4.x, c4.y, c4.z);
+                    }
+
+                    if (g_theTerrainLogic->Get_Ground_Height(c5.x, c5.y, nullptr) + 2.0f >= c5.z) {
+                        points[1].Set(c5.x, c5.y, g_theTerrainLogic->Get_Ground_Height(c4.x, c4.y, nullptr) + 2.0f);
+                    } else {
+                        points[1].Set(c5.x, c5.y, c5.z);
+                    }
+                }
+
+                float innerred;
+                float innergreen;
+                float innerblue;
+                float inneralpha;
+                float outerred;
+                float outergreen;
+                float outerblue;
+                float outeralpha;
+                Get_Color_Components_Real(data->m_innerColor, &innerred, &innergreen, &innerblue, &inneralpha);
+                Get_Color_Components_Real(data->m_outerColor, &outerred, &outergreen, &outerblue, &outeralpha);
+
+                for (unsigned int j = 0; j < data->m_numBeams; j++) {

```

Comment:
the loop starts from numbeams - 1

Reply:
Looking at other binaries the loop goes down in all of them so I have changed it to go down.